### PR TITLE
8 patch article by

### PR DESCRIPTION
**Pull request by alexizumi**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -220,3 +220,54 @@ describe('POST /api/articles/:article_id/comments', () => {
...
+        expect(article[0]).toHaveProperty('article_id', 3);
+        expect(article[0]).toHaveProperty(
+          'title',
+          'Eight pug gifs that remind me of mitch'
+        );
+        expect(article[0]).toHaveProperty('votes', 1);
+        expect(article[0]).toHaveProperty('created_at', expect.any(String));
+        expect(article[0]).toHaveProperty('author', 'icellusedkars');
+        expect(article[0]).toHaveProperty('body', 'some gifs');
+      });
```
Comment (nc-mohammed-dastgir) [inline __tests__/app.test.js:241]: These are good assertions, but article should not be an array here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -220,3 +220,54 @@ describe('POST /api/articles/:article_id/comments', () => {
...
+  test('400: should respond with "Bad request" increment value is not a number', () => {
+    const updateVote = { inc_votes: 'one' };
+    return request(app)
+      .patch('/api/articles/3')
+      .send(updateVote)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad request');
+      });
+  });
```
Comment (nc-mohammed-dastgir) [inline __tests__/app.test.js:272]: Nice error handling tests

File: controllers/app.controller.js [hunk 3]
```diff
@@ -56,3 +57,15 @@ exports.postComment = (req, res, next) => {
...
     .catch(next);
 };
+exports.patchArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+  editArticle(article_id, inc_votes)
+    .then((article) => {
+      if (typeof inc_votes != 'number') {
+        return Promise.reject({ status: 404, msg: 'Bad request' });
+      }
```
Comment (nc-mohammed-dastgir) [inline controllers/app.controller.js:67]: Error handling should not be done in the controller. By entering into this .then(...) we should always be following the happy path

**Review by nc-mohammed-dastgir — COMMENTED** [review]
Nice work, but do have a read of the below comments:

File: __tests__/app.test.js (hunk 1 above)
Comment (alexizumi) [reply __tests__/app.test.js:241]: Corrected

File: __tests__/app.test.js (hunk 2 above)
Comment (alexizumi) [reply __tests__/app.test.js:272]: Thanks

**Pull request merged**